flow/record/adapter/avro.py

```diff
@@ -113,6 +114,8 @@ def __init__(self, path: str, selector: str | None = None, **kwargs):
         }
 
     def __iter__(self) -> Iterator[record.Record]:
+        ctx = get_app_context()
+        selector = self.selector
```

Comment:
The setup for this pattern looks a little awkward, but I recognize it's for performance reasons, so I think it should be fine to keep it like this.